src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java

```diff
@@ -0,0 +1,153 @@
+ * </pre>
+ * 
+ * Sample Configuration:
+ * 
+ * <pre>
+ * <code>
+ * &lt;mailet match="All" class="GuessClassificationMailet"&gt;
+ *    &lt;serviceUrl&gt;http://localhost:9000/email/classification/predict&lt;/serviceUrl&gt;
+ *    &lt;headerName&gt;X-Classification-Guess&lt;/headerName&gt;
+ * &lt;/mailet&gt;
+ * </code>
+ * </pre>
+ * 
+ */
+public class GuessClassificationMailet extends GenericMailet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GuessClassificationMailet.class);
+
+    private static final String SERVICE_URL = "serviceUrl";
+    private static final String HEADER_NAME = "headerName";
+    public static final String HEADER_NAME_DEFAULT_VALUE = "X-Classification-Guess";
+    
+    @VisibleForTesting String serviceUrl;
+    @VisibleForTesting String headerName;
+    private UUIDGenerator uuidGenerator = new UUIDGenerator();
+
+    @Override
+    public void init() throws MessagingException {
+        LOGGER.debug("init GuessClassificationMailet");
+
+        serviceUrl = getInitParameter(SERVICE_URL);
+        LOGGER.debug("serviceUrl value: " + serviceUrl);
+        if (Strings.isNullOrEmpty(serviceUrl)) {
+            throw new MailetException("'serviceUrl' is mandatory");
+        }
+
+        headerName = getInitParameter(HEADER_NAME, HEADER_NAME_DEFAULT_VALUE);
+        LOGGER.debug("headerName value: " + headerName);
+        if (Strings.isNullOrEmpty(headerName)) {
+            throw new MailetException("'headerName' is mandatory");
+        }
+    }
+
+    @Override
+    public String getMailetInfo() {
+        return "GuessClassificationMailet Mailet";
+    }
+
+    @VisibleForTesting void setUUIDGenerator(UUIDGenerator uuidGenerator) {
+        this.uuidGenerator = uuidGenerator;
+    }
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        getClassificationGuess(mail)
+            .ifPresent(classificationGuess -> addHeader(mail, classificationGuess));
+    }
+
+    private Optional<String> getClassificationGuess(Mail mail) {
+        try (CloseableHttpClient httpClient = HttpClients.createDefault();) {
+            HttpPost post = new HttpPost(serviceUrlWithQueryParameters(mail.getRecipients()));
+            post.addHeader("Content-Type", "application/json");
+            post.setEntity(new StringEntity(asJson(mail)));
+            
+            HttpEntity entity = httpClient.execute(post).getEntity();
+            return Optional.ofNullable(IOUtils.toString(entity.getContent(), Charsets.UTF_8));
+            
+        } catch (Exception e) {
+            LOGGER.error("Error occured while contacting classification guess service");
+            return Optional.empty();
+        }
+    }
+
+    private URI serviceUrlWithQueryParameters(Collection<MailAddress> recipients) throws URISyntaxException {
+        URIBuilder uriBuilder = new URIBuilder(serviceUrl);
+        recipients.forEach(address -> uriBuilder.addParameter("recipients", address.asString()));
```

Comment:
I think it's not really a multivaluated header but a header with a comma separated value